File: Week3/Code/get_TreeHeight.py
#!/usr/bin/env python3

"""This function calculates heights of trees given distance of each tree from its base and angle to its top, using the trigonometric formula"""

__appname__ = 'get_TreeHeight.py'
__author__ = 'Talia Al-Mushadani (ta1915@ic.ac.uk)'
__version__ = '0.0.1'
__license__ = "License for this code"

import sys
import math
import csv

def TreeHeight(distance, degrees):
    """function to calculate tree height using trigonometry"""
    radians = degrees * (math.pi/180)
    height = distance * math.tan(radians)
    print("Tree height is: {}".format(height))

    return height

def main(argv=sys.argv):
    """main function of the program"""
    input_csv = open("../Data/{}".format(argv[1]), 'r')
    TreeData = csv.reader(input_csv)
    a = [x for x in TreeData]
    newlist = a[0]
    newlist.append("Height_m")

    for lines in range(1,len(a)):
        line = a[lines]
        height = TreeHeight(float(line[1]), float(line[2]))
        line.append(height)
        newlist.append(line)


    # for i in newlist:
    #     print(i)

    output_file = "../Results/{}".format(argv[1])
    g = open(output_file, 'w')
    csvwrite = csv.writer(g)
    csvwrite.writerow(newlist[0:4])
    for i in newlist[4:]:
        csvwrite.writerow(i)
    g.close()

if __name__ == "__main__": 
    """Makes sure the "main" function is called from command line"""  
    status = main(sys.argv)
    sys.exit("Exit script")

File: Week3/Code/get_TreeHeight.R
#!/usr/bin/env Rscript

#This function calculates heights of trees given distance of each tree from its base and angle to its top, using the trigonometric formula
#
#height = distance * tan(radius)
#
#ARGUMENTS
#input_file: name of input csv file
#degrees: The angle of elevation of tree
#distance: The distance from base of tree (e.g. metres)
#
#OUTPUT
#The heights of the tree, same units as "distance"

# __appname__ = get_TreeHeight.R
# __author__ = Talia Al-Mushadani
# __version__ = 0.0.1
# __license__ = license for this code

rm(list = ls())
graphics.off()

#Function
TreeHeight <- function(degrees, distance){
    #calculates tree height using trig
    radians <- degrees * pi / 180
    height <- distance * tan(radians)
    print(paste("Tree height is:", height))

    return (height)
}

args = commandArgs(trailingOnly = TRUE)
input_csv <- paste('..','Data',args[1], sep = "/")
TreeData <- read.csv(input_csv)
Tree.height.m <- TreeHeight(TreeData$Angle.degrees, TreeData$Distance.m)
TreeDataResult <- cbind(TreeData, Tree.height.m)
output.name <- gsub('.csv', '', args[1])
output.file1 <- paste('..','Results',output.name, sep = "/")
output.file2 <- paste(output.file1,'.csv', sep = "")
write.csv(TreeDataResult, file = output.file2)